### PR TITLE
ORCA-795: Update ORCA graphql internal reconciliation to remove s3 access keys and use IAM logic for DB.

### PR DESCRIPTION
**Pull request by zroehrich**
## Summary of Changes

Removed S3 access keys from graphql code as well as modified graphql task role to allow S3 imports to the V2 Serverless database

Addresses [ORCA-795: Update ORCA graphql internal reconciliation to remove s3 access keys and use IAM logic for DB.](https://bugs.earthdata.nasa.gov/browse/ORCA-795)

## Changes

* Removed `s3_access_key` and `s3_secret_key` from graphql code. Updated policy for graphql task role to allow for S3 imports into the V2 database

## Type of change

Please delete options that are not relevant.

- [ x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Ran in AWS to verify tables from graphql populated and ran unit tests.

## Checklist:

- [x ] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x ] CHANGELOG.md
    - [x ] User documentation
- [ x] My code has passed security scanning
    - [ x] Snyk
    - [ x] git-secrets

### CODE REVIEW

File: tasks/get_current_archive_list/get_current_archive_list.py
Comment (rizbihassan) [inline tasks/get_current_archive_list/get_current_archive_list.py:70]: I don't think graphql is using this code under tasks/get_current_archive_list/get_current_archive_list.py.

It uses https://github.com/nasa/cumulus-orca/blob/develop/graphql/src/adapters/graphql/resolvers/internal_reconcile_report.py#L58

Comment (rizbihassan) [reply tasks/get_current_archive_list/get_current_archive_list.py:70]: We have a ticket https://bugs.earthdata.nasa.gov/browse/ORCA-796 to remove these

**Review by rizbihassan — CHANGES_REQUESTED** [review]
Looks good. So we need to import the graphql role to the v2 cluster, right? 

Once you merge this to 795 branch, that branch should also be deployed and unit tested to double check if terraform throws any error since I removed those access keys from terraform.

Also are you thinking to merge the changes to develop branch?

File: tasks/get_current_archive_list/get_current_archive_list.py
Comment (zroehrich) [reply tasks/get_current_archive_list/get_current_archive_list.py]: Sounds good, reverted those changes

**Comment by zroehrich** [comment]
> Looks good. So we need to import the graphql role to the v2 cluster, right?
> 
> Once you merge this to 795 branch, that branch should also be deployed and unit tested to double check if terraform throws any error since I removed those access keys from terraform.
> 
> Also are you thinking to merge the changes to develop branch?

Yep the PREFIX_orca_gql_tasks_role will be the role attached to the V2 database. Once we test the Terraform and unit tests I think we could merge this to develop unless we want to wait until all the Aurora Serverless work is complete

**Comment by rizbihassan** [comment]
> > Looks good. So we need to import the graphql role to the v2 cluster, right?
> > Once you merge this to 795 branch, that branch should also be deployed and unit tested to double check if terraform throws any error since I removed those access keys from terraform.
> > Also are you thinking to merge the changes to develop branch?
> 
> Yep the PREFIX_orca_gql_tasks_role will be the role attached to the V2 database. Once we test the Terraform and unit tests I think we could merge this to develop unless we want to wait until all the Aurora Serverless work is complete

I thin kwe can merge the changes to develop. I don't see any need for a release soon before the aurora serverless release.

**Pull request merged**
